osutil/group_test.go

```diff
@@ -0,0 +1,74 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+// +build cgo
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package osutil
```

Comment:
(nitpick) we use a slightly different pattern when writing (most) unit tests. The package is called `foo_test` instead of `foo` which forces to use only the public interfaces of the thing that gets tested. This is similar to what the go upstream source is using. If internal access is needed we use a `export_test.go` file that is part of package "foo" to only export things for tests. For this particular test changes should be minimal as it is only testing public functions. So just renaming the package from `osutil` to `osutil_test` should be enough (plus import of osutil and prefixing the methods).